source/extensions/http/cache/file_system_http_cache/cache_file_fixed_block.h

```diff
@@ -159,25 +144,13 @@ class CacheFileFixedBlock {
    * a compile-time constant in ordinary use.
    * @param id The cache version ID to set.
    */
-  void setCacheVersionId(uint32_t id) {
-    contents_.cache_version_id_ = toEndianness<ByteOrder::LittleEndian>(id);
-  }
-
-  union {
-    struct {
```

Comment:
what horrible thing is it that happens if the struct is named? Just having to inject "file_block_." or whatever in references to the fields?

BTW I'm not against using `char raw_[1];` rather than specifying the size as long a you document what magic you are up to with a comment.

I retract my comment about the static-assert now that I see what it's for. That seems fine - sorry for the trouble.